### DISCUSSION TITLE
Include ignored errors in the result

### DISCUSSION THREAD

**Discussion by Luc45** [squizlabs/PHP_CodeSniffer#3811] [Ideas]
It would be great to have a way include ignored errors in the result. For reference, SemGrep adds a `is_ignored` property to each error, it could behave in a similar way.

It could track ignores per-ocurrance on things that it parses, and adding a list of directories/files that are not parsed.

Example JSON result with ignores:

```json
{
  "totals": {
    "errors": 1,
    "warnings": 0,
    "fixable": 1
  },
  "files": {
    "path/to/file.php": {
      "errors": 1,
      "warnings": 0,
      "messages": [
        {
          "message": "Error message",
          "source": "Standard.Category.SniffName",
          "severity": 5,
          "fixable": true,
          "type": "ERROR",
          "line": 10,
          "column": 5
        }
      ]
    }
  },
  "ignored": {
    "totals": {
        "errors": 2,
        "warnings": 1,
    },
    "files": {
      "path/to/file.php": {
        "errors": 1,
        "warnings": 1,
        "fixable": 1,
        "messages": [
          {
            "line": 20,
            "source": "Standard.Category.SniffName",
            "type": "ERROR",
            "message": "Ignored error message",
            "severity": 5,
            "fixable": true,
            "column": 5
          },
          {
            "line": 30,
            "source": "Standard.Category.SniffName",
            "type": "WARNING",
            "message": "Ignored warning message",
            "severity": 5,
            "fixable": false,
            "column": 5
          }
        ]
      },
      "path/to/anotherfile.php": {
        "errors": 1,
        "warnings": 0,
        "messages": [
          {
            "line": 40,
            "source": "Standard.Category.SniffName",
            "type": "ERROR",
            "message": "Ignored error message in a file with ignore comment",
            "severity": 5,
            "fixable": false,
            "column": 5
          }
        ]
      }
    },
    "ignored_files": [
      "path/to/ignoredfile.php"
    ],
    "ignore_patterns": [
      {
        "path": "vendor/**",
        "source": ""
      },
      {
        "path": "path/to/ignored/directory/**",
        "source": "Standard.Category.SniffName"
      }
    ]
  }
}
```

- Entries under `ignored.files` are those ignored by local comments such as `// phpcs:ignore, // phpcs:ignore <rule>, // phpcs:disable, // phpcs:disable <rule>`.
- Files in `ignored.ignored_files` are those ignored by `// phpcs:ignoreFile`
- In `ignored.ignore_patterns`, we could include paths that were ignored by `--ignore` or `exclude-pattern`

**Comment by jrfnl** [comment]
I don't see the point of this. What's the use case ?

Ignored issues are ignored, which is exactly the intended behaviour. If for some reason - like reviewing ignore annotations - you want to ignore the ignore annotations, you can do so for a run by using `--ignore-annotations`.

**Reply by Luc45** [reply]
We are using PHPCS in a QA tool to scan code we don't control. We want to allow developers to ignore certain false positives, but we want to keep track of what's ignored to make sure developers aren't just ignoring errors instead of fixing them.

**Reply by jrfnl** [reply]
In that case, I suppose you can create the report you desire by running PHPCS twice, once without `--ignore-annotations`, once with `--ignore-annotations` and doing a diff and an intersect of the output of both runs.

**Reply by Luc45** [reply]
I didn't know about `--ignore-annotations`, that will work for me, thanks. Should we close this or keep it open to see if more people would like this feature?

**Reply by jrfnl** [reply]
I can't speak for the repo, but if you want my opinion: this is so edge-case, that it would be unreasonable to expect maintainers of PHPCS to spend significant time on creating and maintaining it, so I would very quickly vote against the feature.
